spec/unit/intercom/traits/api_resource_spec.rb

```diff
@@ -107,12 +122,28 @@ class ConcreteApiResource; include Intercom::Traits::ApiResource; end
 
     api_resource.from_hash(object_hash)
     initialized_api_resource = ConcreteApiResource.new(object_hash)
-
-    except(object_json, 'type').keys.each do |attribute|
+    except(object_json, 'type', 'nested_fields').keys.each do |attribute|
```

Comment:
Temporarily added.
- Typed objects are created only when the hash has a key `type` as string. Symbol is not taken into account. 

Even though this test passes. It should've ideally failed like the previous one.
Added `nested_fields` here just to maintain consistency.

Reply:
Will be addressed in a separate PR, Possible solutions
- Manual check of type as string/symbol
- Hash.symbolize_keys
- Hash.with_indifferent_access